ubcpi/answer_pool.py

```diff
@@ -0,0 +1,315 @@
+import random
+import persistence as sas_api
+
+POOL_SIZE = 50
+
+
+class UnknownChooseAnswerAlgorithm(Exception):
+    pass
+
+
+def offer_answer(pool, answer, rationale, student_id, algo):
+    """
+    submit a student answer to the answer pool
+
+    The answer maybe selected to stay in the pool depending on the selection algorithm
+
+    Args:
+        pool (dict): answer pool
+            Answer pool format:
+            {
+                option1_index: {
+                    'student_id': { can store algorithm specific info here },
+                    ...
+                }
+                option2_index: ...
+            }
```

Comment:
Are you saying that the data structure would look like this:

{0: {37: {}, 42: {}}, 1: {56: {}, 81: {}}, 2: {71: {}, 19: {}}}

where the keys 0, 1, and 2 are problem choice identifiers, and 37, 42, 56, etc are student ID's.

What I'm wondering is if a 3 level nested dictionary is a good idea. @nedbat ?